samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -51,21 +53,17 @@ static K_SEM_DEFINE(sem_security_updated, 0, 1);
 static K_SEM_DEFINE(sem_sinks_discovered, 0, 1);
 static K_SEM_DEFINE(sem_sources_discovered, 0, 1);
 static K_SEM_DEFINE(sem_stream_configured, 0, 1);
-static K_SEM_DEFINE(sem_stream_qos, 0, 1);
+static K_FIFO_DEFINE(fifo_stream_qos);
```

Comment:
Do you mean calling `k_sem_init(sem, initial_count, limit)` before QoS instead? I was unsure it's allowed to call this API function multiple times on the same semaphore object.

Reply:
I was more thinking about doing
```c
static K_SEM_DEFINE(sem_stream_qos, 0, ARRAY_SIZE(sinks) + ARRAY_SIZE(sinks));
```

and then where you do 
```c
k_fifo_put(&fifo_stream_qos, stream);
``` 
you just revert to 
```c
k_sem_give(&sem_stream_qos);
```

and where you do 
```c
k_fifo_get(&fifo_stream_qos, K_FOREVER);
```
you just do

```c
k_sem_take(sem_stream_qos);
```

It will have the same effect as the FIFO, but just using a semaphore instead. Whether or not it is better, is a good question :)